### PR TITLE
feat(resource): upload audio (DSP-1799)

### PR DESCRIPTION
**Pull request by kilchenmann**
resolves DSP-1799

~~Upload audio file works, but we do not have an audio player yet; this will be implemented in the next iteration. It's the old chicken or the egg dilemma. It means for the reviewers, when you want to test it (even it's a code review only), you have to check the API answer in the developer tools or you should find the file in the `dsp-api/sipi/images/[project]/`-folder~~

### CODE REVIEW

**Review by mdelez — CHANGES_REQUESTED** [review]
This works well if I use it exactly how I'm supposed to but I noticed that if I upload a file (via drag and drop) and then delete it and click save, it scrolls to the upload section. If I then upload a file and click save I get this error and no matter what I do I cannot save the resource:
Screen Shot 2021-07-22 at 10 50 26

Another thing I noticed is that if I edit my data model, the changes aren't reflected immediately if I click on the plus button on the top right to create a new resource. But maybe this could be a separate PR.
Screen Shot 2021-07-22 at 10 44 23

**Comment by kilchenmann** [comment]
> This works well if I use it exactly how I'm supposed to but I noticed that if I upload a file (via drag and drop) and then delete it and click save, it scrolls to the upload section. If I then upload a file and click save I get this error and no matter what I do I cannot save the resource:

--> I'm able to reproduce the issue, but I don't get the error message. But I figured out, that the submit button is not disabled anymore in case the form is wrong / has missing values in required fields 

> Another thing I noticed is that if I edit my data model, the changes aren't reflected immediately if I click on the plus button on the top right to create a new resource. But maybe this could be a separate PR.

--> this is indeed a separate task. Would you like to write a bug report in youtrack and assign to me?

**Comment by mdelez** [comment]
> > This works well if I use it exactly how I'm supposed to but I noticed that if I upload a file (via drag and drop) and then delete it and click save, it scrolls to the upload section. If I then upload a file and click save I get this error and no matter what I do I cannot save the resource:
> 
> --> I'm able to reproduce the issue, but I don't get the error message. But I figured out, that the submit button is not disabled anymore in case the form is wrong / has missing values in required fields
> 
> > Another thing I noticed is that if I edit my data model, the changes aren't reflected immediately if I click on the plus button on the top right to create a new resource. But maybe this could be a separate PR.
> 
> --> this is indeed a separate task. Would you like to write a bug report in youtrack and assign to me?

1. Yes, ideally the save button should be disabled if the form is invalid but I wasn't sure if this was set up already
2. Bug report opened -> DSP-1810

File: src/app/workspace/resource/representation/upload/upload.component.ts [hunk 1]
```diff
@@ -283,6 +295,9 @@ export class UploadComponent implements OnInit {
             case 'document':
                 this.allowedFileTypes = this.supportedDocumentTypes;
                 break;
+            case 'audio':
+                this.allowedFileTypes = this.supportedAudioTypes;
+                break;
             default:
```
Comment (Vijeinath) [inline src/app/workspace/resource/representation/upload/upload.component.ts:301]: Not severe: In this file there are incongruent blank lines within switch blocks. Either use a blank line after every case block or no blank line after case block

**Review by Vijeinath — APPROVED** [review]
I'm happy with it

File: src/app/workspace/resource/representation/upload/upload.component.ts (hunk 1 above)
Comment (kilchenmann) [reply src/app/workspace/resource/representation/upload/upload.component.ts]: resolved in 8f6c0e4

**Comment by kilchenmann** [comment]
> 1. Yes, ideally the save button should be disabled if the form is invalid but I wasn't sure if this was set up already

--> should be resolved now

**Review by mdelez — APPROVED** [review]
Save button works appropriately now :)

**Pull request merged**